File: algorithms-data-structures/ts/src/algorithms/mixed/group-anagrams/group.test.ts
import {test, expect} from "bun:test";
import {groupAnagrams} from "./group";

test("should group anagrams from a list of words", () => {
  let input = ["eat", "tea", "tan", "ate", "nat", "bat"];
  let output = [["eat", "tea", "ate"], ["tan", "nat"], ["bat"]];
  let got = groupAnagrams(input);
  console.log("got", got);
  expect(got).toEqual(output);
});

test("should handle single empty string as input", () => {
  let input = [""];
  let output = [[""]];
  expect(groupAnagrams(input)).toEqual(output);
});

test("should handle single character string as input", () => {
  let input = ["a"];
  let output = [["a"]];
  expect(groupAnagrams(input)).toEqual(output);
});

test("should group multiple empty strings together", () => {
  let input = ["", ""];
  let output = [["", ""]];
  expect(groupAnagrams(input)).toEqual(output);
});

File: go/src/algorithms/mixed/group-anagrams/group_test.go
package groupanagrams

import (
	"testing"

	"github.com/stretchr/testify/assert"
)

func TestGroupAnagrams(t *testing.T) {
	input := []string{"eat", "tea", "tan", "ate", "nat", "bat"}
	res := groupAnagrams(input)

	assert.Equal(t, 3, len(res))
	assert.Equal(t, []string{"eat", "tea", "ate"}, res[0])

	assert.Equal(t, []string{"tan", "nat"}, res[1])
	assert.Equal(t, []string{"bat"}, res[2])

	input = []string{"a"}
	res = groupAnagrams(input)
	assert.Equal(t, 1, len(res))
	assert.Equal(t, []string{"a"}, res[0])

}
